UpdatedTicketHubDDL.sql

```diff
@@ -0,0 +1,173 @@
+CREATE TABLE User (
+    UserID INT PRIMARY KEY AUTO_INCREMENT,
+    Username VARCHAR(50) UNIQUE,
+    FirstName VARCHAR(50),
+    LastName VARCHAR(50),
+    Birthdate DATE,
+    Password VARCHAR(100),
+    Email VARCHAR(100) UNIQUE
+);
+
+CREATE TABLE Buyer (
+    BuyerID INT PRIMARY KEY,
+    UserID INT UNIQUE, 
+    FOREIGN KEY (UserID) REFERENCES User(UserID)
+);
+
+CREATE TABLE Admin (
+    AdminID INT PRIMARY KEY,
+    UserID INT UNIQUE,
+    FOREIGN KEY (UserID) REFERENCES User(UserID)
+);
+
+CREATE TABLE Seller (
+    SellerID INT PRIMARY KEY,
+    UserID INT UNIQUE,
+    FOREIGN KEY (UserID) REFERENCES User(UserID)
+);
+CREATE TABLE TicketInventory (
+    TicketID INT PRIMARY KEY,
+    SellerID INT,
+    TicketQty INT,
+    TicketStatus BOOLEAN,
+    IsOverdue BOOLEAN,
+    FOREIGN KEY (SellerID) REFERENCES Seller(SellerID)
+);
+
+CREATE TABLE Event (
+    EventID INT PRIMARY KEY,
+    EventName VARCHAR(100),
+    EventDescription VARCHAR(255),
+    EventType VARCHAR(50),
+    ArtistID INT,
+    Location VARCHAR(100),
+    DateTime DATETIME
+);
```

Comment:
I don't know that the values in the event table are consistent with what we have set up in the interface. Right now, on the seller side, the seller is able to edit the event name, location and date time.

Reply:
- I don't think we need UserId in the tables: buyer, seller, admin.
- Therefore, the foreign keys of each table should be not the 'UserId' but 'Buyer/Seller/AdminId'
- It would be great if you could set the length constraint for each Id attributes as well
- I don't think we are supposed to have ArtistId anymore